restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1008,13 +1022,16 @@ private ExecutionContext nestedPopulate(ExecutionContext context) {
                         .from(keyspace, collection)
                         .where(keyCondition)
                         .where(buildConditions(nestedCondition, db.treatBooleansAsNumeric()))
+                        .limit(1)
```

Comment:
this seems smart, I need to pull these changes in my reactive flow after this is merged.